### PR TITLE
 Add test case for issue https://github.com/wso2/wso2-axis2/issues/123

### PR DESCRIPTION
**Pull request by vinok88**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java [hunk 1]
```diff
@@ -48,6 +52,17 @@ public void callMediatorBlockingDirectEndpointTest() throws AxisFault {
         assertTrue(responseContainsWSO2);
     }
 
+    @Test(groups = { "wso2.esb" },
+          description = "Invoke a backend that returns a 204, empty response")
+    public void callMediatorBlockingNoContentResponseTest() throws IOException {
+        SimpleHttpClient client = new SimpleHttpClient();
+        Map headers = new HashMap(1);
+        headers.put("Content-Type", "application/json");
+        org.apache.http.HttpResponse response = client.doGet("http://localhost:8480/CallMediatorNoContentTestAPI/",
```
Comment (chanikag) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java]: Can't we use getApiInvocationURL() method without hard coding the host and port

File: integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentBackendAPI.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<api xmlns="http://ws.apache.org/ns/synapse" name="CallMediatorNoContentBackendAPI" context="/CallMediatorNoContentBackendAPI">
```
Comment (chanikag) [inline integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentBackendAPI.xml]: Add license headers?

File: integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentTestAPI.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<api xmlns="http://ws.apache.org/ns/synapse" name="CallMediatorNoContentTestAPI" context="/CallMediatorNoContentTestAPI">
```
Comment (chanikag) [inline integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentTestAPI.xml]: Add license header?

File: integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentBackendAPI.xml (hunk 2 above)
Comment (vinok88) [reply integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentBackendAPI.xml]: Do we really need licencing for configs?  I checked few other configs and most of them do not have license headers.

File: integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentTestAPI.xml (hunk 3 above)
Comment (vinok88) [reply integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentTestAPI.xml]: Do we really need licencing for configs?  I checked few other configs and most of them do not have licence headers.

Comment (chanikag) [reply integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentTestAPI.xml]: Yes, all files which can be commented should have the license header.

File: integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentBackendAPI.xml (hunk 2 above)
Comment (chanikag) [reply integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentBackendAPI.xml]: Yes, all files which can be commented should have the license header.

Comment (vinok88) [reply integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentBackendAPI.xml]: Fixed. Thanks.

File: integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentTestAPI.xml (hunk 3 above)
Comment (vinok88) [reply integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/CallMediatorNoContentTestAPI.xml]: Fixed. Thanks.

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java (hunk 1 above)
Comment (vinok88) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingDirectEndpointTestCase.java]: Fixed. Thanks.

**Pull request merged**